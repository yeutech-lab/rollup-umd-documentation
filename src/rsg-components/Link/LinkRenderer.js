import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';
import Styled from 'rsg-components/Styled';

const styles = ({ color }) => ({
	link: {
		'&, &:link, &:visited': {
			fontSize: 'inherit',
			color: color.link,
			textDecoration: 'none',
		},
		'&:hover, &:active': {
			isolate: false,
			color: color.linkHover,
			cursor: 'pointer',
		},
	},
});

export function LinkRenderer({ classes, children, ...props }) {
	return (
		<a {...props} className={cx(classes.link, props.className)}>
			{children}
		</a>
	);
}

LinkRenderer.propTypes = {
	children: PropTypes.node,
	className: PropTypes.string,
	classes: PropTypes.object.isRequired,
};

export default Styled(styles)(LinkRenderer);
