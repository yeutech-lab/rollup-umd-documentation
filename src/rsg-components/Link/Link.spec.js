import React from 'react';
import { LinkRenderer } from './LinkRenderer';

const href = '/foo';
const children = 'Foo';

it('renderer should render link', () => {
	const actual = shallow(
		<LinkRenderer href={href} classes={{}}>
			{children}
		</LinkRenderer>
	);

	expect(actual).toMatchSnapshot();
});

it('should compose passed class names', () => {
	const actual = shallow(
		<LinkRenderer classes={{ link: 'baseLinkClass' }} href={href} className="customClass">
			{children}
		</LinkRenderer>
	);

	expect(actual.find('a').prop('className')).toBe('baseLinkClass customClass');
});
