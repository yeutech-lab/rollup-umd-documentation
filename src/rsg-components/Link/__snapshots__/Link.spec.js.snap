// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderer should render link 1`] = `
<a
  className=""
  href="/foo"
>
  Foo
</a>
`;
